source/Body.cpp

```diff
@@ -328,6 +683,80 @@ void Body::PauseAnimation()
 
 
 
+void Body::ShowDefaultSprite(bool defaultSprite)
+{
+	this->returnDefaultSprite = defaultSprite;
+}
+
+
+
+// Check the trigger sprites for the current state and initiate state transition for that trigger.
+// E.g. if FIRING a certain weapon, check if we have an animation for that weapon, and trigger it.
+void Body::CheckTriggers()
+{
+	for(int i = 0; i < BodyState::NUM_STATES; i++)
```

Comment:
Could we use a C++11 loop here?

Reply:
Possible, but I would have to create an array of the elements of the enum as a const. If that is something that is preferred according to the code style, I am happy to make that change.